Skyra.Core/Cache/Models/Prompts/CoreRichDisplay.cs

```diff
@@ -0,0 +1,315 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using JetBrains.Annotations;
+using Newtonsoft.Json;
+using Skyra.Core.Utils;
+using Spectacles.NET.Types;
+
+namespace Skyra.Core.Cache.Models.Prompts
+{
+	public sealed class CoreRichDisplay : ICorePromptState, ICorePromptStateReaction
+	{
+		public CoreRichDisplay(ulong authorId, ulong messageId)
+		{
+			AuthorId = authorId;
+			MessageId = messageId;
+			InternalTemplate = new CoreMessageEmbed();
+			InformationPage = null;
+			AllowedEmojis = new (CoreRichDisplayReactionType, string)[0];
```

Comment:
Why are these being defaulted to an empty array?

Reply:
Can I use `default` instead? :thinking:

Reply:
default for reference types (arrays are) is null. So it depends if that's what you want

Reply:
The reason why it's not a nullable array is to make working on it much easier.

Nevertheless you reminded me I need to make it a List (check AddPage, it screams inefficiency.